src/block/BaseSign.php

```diff
@@ -114,24 +112,22 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 				return false;
 			}
 
-			$lines = array_map(static fn(string $line) => self::getColorCode($dyeColor) . TextFormat::clean($line), $this->getText()->getLines());
-
-			$ev = new SignChangeEvent($this, $player, new SignText($lines));
-			$ev->call();
+			$this->textColor = $dyeColor;
 
-			if(!$ev->isCancelled()){
-				$this->textColor = $dyeColor;
-				$this->glowingText = true;
+			$this->position->getWorld()->setBlock($this->position, $this);
 
-				$this->setText($ev->getNewText());
+			$item->pop();
+			return true;
 
+		}elseif($item->getId() === ItemIds::DYE && $item->getMeta() === 0){
```

Comment:
Please leave a comment indicating what this is. I know there isn't currently a better way to identify the thing, but it's quite confusing to read nonetheless.